azkaban/util.py

```diff
@@ -69,20 +69,27 @@ class Config(object):
 
   """
 
-  def __init__(self, path=None):
+  def __init__(self, path=None, max_read_cnt=10):
     self.parser = RawConfigParser()
     self.path = path or expanduser('~/.azkabanrc')
     # TODO: make the default path be configurable via an environment variable.
     if exists(self.path):
-      try:
-        self.parser.read(self.path)
-      except ParsingError:
-        raise AzkabanError('Invalid configuration file %r.', self.path)
+      read_cnt = 0
```

Comment:
IIUC the issue is with `save` calls happening concurrently with reads. If so, I think we can fix this by only saving when it is required. When the config file only has "new-style" aliases (which should be all the time, except for a one time conversion) no writing is required, so we can probably get away without any retry logic.

Reply:
@mtth 
Thanks for reviewing this PR.

> IIUC the issue is with save calls happening concurrently with reads.

Yes, that's right. We can fix this issue without this retry logic. The only case a parser comes to read an empty file is that a parser's read process and save/write process of Azkaban session id happen simultaneously. I'm wondering we should implement this retry logic to handle this case or not.  I know implementing this kind of retry logic is a little bit messy and I'd like to know your idea.